controllers/bootstrap/init.go

```diff
@@ -1158,11 +1189,113 @@ func (b *Bootstrap) DeployCertManagerCR() error {
 	return nil
 }
 
+// CleanNamespaceScopeResources will delete the v3 NamesapceScopes resources and namespace scope operator
+// NamespaceScope resources include common-service, nss-managedby-odlm, nss-odlm-scope, and odlm-scope-managedby-odlm
+func (b *Bootstrap) CleanNamespaceScopeResources() error {
+	// Get the common-service OperandRegistry
+	operandRegistry, err := b.GetOperandRegistry(ctx, constant.MasterCR, b.CSData.ServicesNs)
+	if err != nil && operandRegistry == nil {
+		klog.Errorf("Failed to get common-service OperandRegistry: %v", err)
+		return err
+	}
+
+	// List all requested operators
+	if operandRegistry.Status.OperatorsStatus != nil {
+		for operator := range operandRegistry.Status.OperatorsStatus {
+			// If there is a requested operator's installMode is "no-op", then skip call delete function
+			for _, op := range operandRegistry.Spec.Operators {
+				if op.Name == operator && op.InstallMode == "no-op" {
+					klog.Infof("The v3 operator %s is still requested, skip cleaning the resources", operator)
+					return nil
+				}
+			}
+		}
+	}
+
+	// If the topology is Simple or All Namespaces Mode, delete all NamespaceScope resources
+	if b.CSData.WatchNamespaces == "" || len(strings.Split(b.CSData.WatchNamespaces, ",")) == 1 {
+
+		klog.Info("Cleaning NamespaceScope resources in Simple Topology or All Namespaces Mode")
+		// Delete v3 Namespace Scope operator
+		klog.Infof("Uninstall v3 Namespace Scope operator in servicesNamespace %s when the topology is Simple or All Namespaces Mode", b.CSData.ServicesNs)
```

Comment:
Could we move those logs after we verified it is v3.x NamespaceScope Operator is installed, not v4?
Otherwise, it would be confusing that why CS operator states `cleaning NSS resources in each reconciliation loop.

```golang
if strings.HasPrefix(sub.Spec.Channel, "v4.") {
    klog.Infof("The %s subscription is in the v4.x channel, skip cleaning up", constant.NsSubName)
    return nil
}
klog.Info("Cleaning NamespaceScope resources in Simple Topology or All Namespaces Mode")
klog.Infof("Uninstall v3 Namespace Scope operator in servicesNamespace %s when the topology is Simple or All Namespaces Mode", b.CSData.ServicesNs)

if err := b.DeleteOperator(constant.NsSubName, b.CSData.ServicesNs); err != nil {
  klog.Errorf("Failed to uninstall v3 Namespace Scope operator in servicesNamespace %s", b.CSData.ServicesNs)
  return err
}
```